chain/indexer/src/streamer/mod.rs

```diff
@@ -60,16 +60,22 @@ async fn build_streamer_message(
         let mut outcomes = shards_outcomes
             .remove(&header.shard_id)
             .expect("Execution outcomes for given shard should be present");
-
-        let mut indexer_transactions: Vec<IndexerTransactionWithOutcome> = vec![];
-        for transaction in transactions {
-            let outcome_position = outcomes.iter().position(|outcome| outcome.execution_outcome.id == transaction.hash)
-                .expect("The transaction execution outcome should always present in the same block as the transaction itself");
-            indexer_transactions.push(IndexerTransactionWithOutcome {
-                outcome: outcomes.remove(outcome_position),
-                transaction,
+        let mut receipt_outcomes = outcomes.split_off(transactions.len());
+
+        let indexer_transactions = transactions
+            .iter()
+            .enumerate()
+            .map(|(index, transaction)| {
+                assert_eq!(
+                    outcomes[index].execution_outcome.id, transaction.hash,
```

Comment:
Use [`zip`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.zip) instead of `enumerate` + fetching by index.